src/Microsoft.VisualStudio.Validation/Assumes.cs

```diff
@@ -48,7 +50,7 @@ public static void NotNullOrEmpty([ValidatedNotNull, NotNull]string value)
         /// </summary>
         /// <typeparam name="T">The type of value to test.</typeparam>
         [DebuggerStepThrough]
-        public static void NotNullOrEmpty<T>([ValidatedNotNull, NotNull]ICollection<T> values)
+        public static void NotNullOrEmpty<T>([ValidatedNotNull, NotNull]ICollection<T>? values)
```

Comment:
We should allow for collections containing null items.

```suggestion
        public static void NotNullOrEmpty<T>([ValidatedNotNull, NotNull]ICollection<T?>? values)
```

Reply:
We do. Since `T` has no `T : class` constraint we can't make it `T?` and C# considers it implicitly to be `T?` at the caller's side.